### PR TITLE
Add bash completion script

### PR DESCRIPTION
**Pull request by russellkan**
Addressing Issue #2 

Done:
- Add bash completion script to complete commands `list`, `run`, `show`, and `runshow`
- Add script to auto-install with `setuptools` in `setup.py`
- Dynamically pull suites and cases when completing commands requiring `casespec` or `resultspec`

To do:
- Enable script by sourcing it automatically during install if possible
- Dynamically pull commands instead of hard coding a list (optional)
- Test the script using `pytest` (optional)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
     },
 
     package_data={},
-    data_files=[],
+    data_files=[('/etc/bash_completion.d/', ['hdat/hdat_completion'])],
```
Comment (johndgiese) [inline setup.py]: I looked on my Mac, and there isn't a directory called `/etc/bash_complete.d`.  It seems that there is `/usr/local/etc/bash_complete.d`.

I suspect the right folder will change from distribution to distribution...

This seems like a hard problem.  See https://docs.pytest.org/en/latest/bash-completion.html and argcomplete docs for some other ways people have solved it.

Perhaps we can install the file in `~/bash_complete.d/` instead?

Will setup.py create the directory if it is missing?

File: hdat/hdat_completion [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# To enable this bash completion, source this file:
+#
+# source /etc/bash_completion.d/hdat_completion
```
Comment (johndgiese) [inline hdat/hdat_completion]: Nice documentation here :)

File: setup.py (hunk 1 above)
Comment (russellkan) [reply setup.py]: That makes sense. There were a few examples that I found on GitHub where some of them had documentation telling people to install it in their local bash completion folder or add it into `.bashrc`.

I tested putting `~/bash_completion.d/` into `setup.py` but it ended up putting it within the `build` folder that was created by `setuptools`.

`Setup.py` will indeed create a directory if it does not already exist, so I changed the target location to `/usr/local/etc/bash_completion.d/`. Either way works; which one would you suggest be used?

Comment (johndgiese) [reply setup.py]: Lets go with `~/bash_completion.d/`.  I think it is safer to touch this since it is in the user's home directory.

**Comment by johndgiese** [comment]
Great work! I am excited to use the auto complete!

File: setup.py (hunk 1 above)
Comment (russellkan) [reply setup.py]: How do I access the user's home directory with `setuptools`? When I use this:
```
data_files=[('~/bash_completion.d/', ['hdat/hdat_completion'])]
```
it installs it from the current working directory into:
```
creating build/bdist.linux-x86_64/egg/~/bash_completion.d
copying hdat/hdat_completion -> build/bdist.linux-x86_64/egg/~/bash_completion.d/
```

Comment (johndgiese) [reply setup.py]: Good question! I have no idea.

Comment (russellkan) [reply setup.py]: I figured it out! Adding a backslash before the tilde worked.
```
data_files=[('/~/bash_completion.d/', ['hdat/hdat_completion'])]
```

Comment (russellkan) [reply setup.py]: Correction: needed to use `path.expanduser('~')` to get it to properly pull the home directory.

New commit should work properly regardless of system.

File: setup.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 from os import path
 
 here = path.abspath(path.dirname(__file__))
+completion_dir = '/'.join([path.expanduser('~'), 'bash_completion.d/'])
```
Comment (johndgiese) [inline setup.py]: You may want to use `os.path.join` instead of `"/".join`, because on some operating systems "\" should be used instead.  :(

**Pull request merged**